api/app.go

```diff
@@ -1200,16 +1200,25 @@ func setEnv(w http.ResponseWriter, r *http.Request, t auth.Token) (err error) {
 }
 
 func IsEnvVarValid(envs ...apiTypes.Env) error {
+	envNames := make([]string, len(envs))
+	for i, e := range envs {
+		envNames[i] = e.Name
+	}
+
+	return isEnvsNamesVarValid(envNames...)
+}
+
+func isEnvsNamesVarValid(envs ...string) error {
 	var errs errors.MultiError
 
 	for _, e := range envs {
-		if isInternalEnv(e.Name) {
-			errs.Add(fmt.Errorf("cannot change an internal environment variable (%s): %w", e.Name, apiTypes.ErrWriteProtectedEnvVar))
+		if isInternalEnv(e) {
```

Comment:
I would extract it to a function
```suggestion
func isValid(env string) error {
	if isInternalEnv(env) {
		return fmt.Errorf("cannot change an internal environment variable (%s): %w", env, apiTypes.ErrWriteProtectedEnvVar)
	}

	if err := isEnvVarName(env); err != nil {
		return fmt.Errorf("%q is not valid environment variable name: %w", env, err)
	}

	return nil
}
```

Reply:
I would also change the name isEnvVarName to isUnixLike